### DISCUSSION TITLE
.GeneratePdfAndShow() not working on server

### DISCUSSION THREAD

**Discussion by kjennes** [QuestPDF/QuestPDF#953] [Q&A] [answered]
Hi, I have developed an application in .net core with razor pages to create my invoices. Locally it all works well, I can use both document.GeneratePDF(filename) and document.GeneratePDFAndShow() for a preview. But when I deploy my solution to my webserver, I can still use document.GeneratePDF(filename) but document.GeneratePDFAndShow() gives me the error "An error occurred trying to start process 'D:\HostingSpaces\myName\myServer\wwwroot\wwwroot\invoices_tmp\QuestPDF Document 1.pdf' with working directory 'D:HostingSpaces\myName\myServer\wwwroot\wwwroot\invoices_tmp'. Access is denied." What am I doing wrong? I can see the file QuestPDF Document 1.pdf has been created in that folder, so I am confused why I gett an Access is denied error? All help much appreciated!

**Comment by MarcinZiabek** [comment, marked as answer]
The `document.GeneratePDFAndShow` method generates a PDF file, saves it in a temporary location and then attempts to open it using a default PDF viewer in your operating system. Using this method does not make sense in the server environment.

You likely want to generate the PDF file as a byte array and return it as a file using the ASP.NET API in some controller's endpoint.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
